src/dtls/mod.rs

```diff
@@ -317,6 +321,184 @@ impl Dtls {
     }
 }
 
+/// Keying material used as master key for SRTP.
+pub(crate) struct KeyingMaterial(Vec<u8>);
+
+impl KeyingMaterial {
+    #[cfg(any(test, feature = "_internal_test_exports"))]
+    pub(crate) fn new(m: &[u8]) -> Self {
+        KeyingMaterial(m.into())
+    }
+}
+
+impl Deref for KeyingMaterial {
+    type Target = [u8];
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl std::fmt::Debug for KeyingMaterial {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "KeyingMaterial")
+    }
+}
+
+#[derive(Debug, Clone)]
+pub(crate) enum Certificate {
+    #[cfg(feature = "openssl")]
+    Openssl(ossl::DtlsCert),
+}
+
+impl Certificate {
+    #[cfg(feature = "openssl")]
+    pub(crate) fn new_openssl() -> Certificate {
+        Certificate::Openssl(ossl::DtlsCert::new())
+    }
+}
+
+enum Context {
```

Comment:
`Context` is too generic here. I think we need `DtlsContext` or maybe `TlsContext`

Reply:
I mean, we are within the `dtls` module here and this is private so it is a bit repetitive but I'll rename it :)

Reply:
Ha! :)